homeassistant/components/cast/media_player.py

```diff
@@ -199,10 +199,6 @@ async def async_added_to_hass(self):
     async def async_will_remove_from_hass(self) -> None:
         """Disconnect Chromecast object when removed."""
         await self._async_disconnect()
-        if self._cast_info.uuid is not None:
-            # Remove the entity from the added casts so that it can dynamically
-            # be re-added again.
-            self.hass.data[ADDED_CAST_DEVICES_KEY].remove(self._cast_info.uuid)
```

Comment:
Is this cleaned up somewhere else?

Reply:
No, it's not.
If we add support for reloading the cast integration, it should be cleaned up when the integration is unloaded so entities can be re-discovered when it's loaded again.